kdb/keyset_test.go

```diff
@@ -116,6 +124,8 @@ func TestRemoveKey(t *testing.T) {
 
 	foundKey = ks.Lookup(k2)
 	Assertf(t, foundKey == nil, "KeySet contains key %s", k2.Name())
+
+	// TODO REVIEW: How to pop via ksLookup? Or do you have Pop with argument? (which is maybe nicer API)
```

Comment:
That would be the `Remove` function

Reply:
Ok nice. Wouldn't a RemoveByName make sense? Did you compare how much more expensive a key creation in Go compared to key creation inside ksLookupByName is?

Reply:
What do you mean with key creation in Go? Creating a key and passing it to `ksLookup` instead of calling `ksLookupByName`?

Reply:
I've added the `RemoveByName` function

Reply:
> What do you mean with key creation in Go?

As you already work on comparing the iterators, I think you can drop this idea for now.